cmd/httpd/handlers/translations.go

```diff
@@ -0,0 +1,210 @@
+	LanguagesHelper   data.LanguagesHelper
+}
+
+// Traduction represents the traduction of a word
+type Traduction struct {
+	SourceLanguage string             `json:"source_language"`
+	TargetLanguage string             `json:"target_language"`
+	SourceWord     primitive.ObjectID `json:"source_word"`
+	TargetWord     primitive.ObjectID `json:"target_word"`
+}
+
+func (h *translations) list(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	var response web.Response
+
+	count, error := h.TranslationHelper.Count(r.Context())
+
+	if error != nil {
+		web.RespondWithInternal(r.Context(), w)
+		return
+	}
+
+	pagination := web.GetPagination(r)
+	links, error := web.MakePaginationLinks(*r.URL, count, pagination)
+
+	if error != nil {
+		web.RespondWithInternal(r.Context(), w)
+		return
+	}
+
+	options := data.FindOptions{
+		Filter: map[string]interface{}{},
+		Limit:  pagination.Size,
+		Skip:   pagination.Number - 1,
+	}
+
+	list, error := h.TranslationHelper.Find(r.Context(), options)
+
+	if error != nil {
+		web.RespondWithInternal(r.Context(), w)
+		return
+	}
+
+	response.Data = list
+	response.Links = links
+	response.Meta = map[string]interface{}{
+		"totalPages": math.Ceil(float64(count) / float64(pagination.Size)),
+	}
+
+	web.RespondWith(r.Context(), w, http.StatusOK, response)
+}
+
+func (h *translations) create(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	var newTranslation data.NewTranslation
+	var response = web.Response{}
+
+	errors := web.Decode(r, &newTranslation)
+
+	if len(errors) > 0 {
+		response.Errors = errors
+		web.RespondWith(r.Context(), w, http.StatusBadRequest, response)
+		return
+	}
+
+	translation, error := h.TranslationHelper.New(r.Context(), newTranslation)
+
+	if error != nil {
+		web.RespondWithInternal(r.Context(), w)
+		return
+	}
+
+	response.Data = translation
+
+	web.RespondWith(r.Context(), w, http.StatusOK, response)
+}
+
+func (h *translations) get(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	var response = web.Response{}
+	var word = ps.ByName("id")
+
+	sourceLanguageID, exist := r.URL.Query()["source"]
+	targetLanguageID, exist := r.URL.Query()["target"]
+
+	if !exist || len(sourceLanguageID) == 0 {
+		response.Errors = web.Error{
+			Status: 400,
+			Source: web.ErrorSource{
+				Pointer: "source language is missing",
+			},
+			Detail: "Missing values",
+		}
+		web.RespondWith(r.Context(), w, http.StatusBadRequest, response)
+		return
+	}
+
+	if !exist || len(targetLanguageID) == 0 {
+		response.Errors = web.Error{
+			Status: 400,
+			Source: web.ErrorSource{
+				Pointer: "target language is missing",
+			},
+			Detail: "Missing values",
+		}
+		web.RespondWith(r.Context(), w, http.StatusBadRequest, response)
+		return
+	}
+
+	ID, error := primitive.ObjectIDFromHex(word)
+
+	if error != nil {
+		web.RespondWithBadRequest(r.Context(), w)
+		return
+	}
+
+	sourceWord, error := h.WordsHelper.FindByID(r.Context(), sourceLanguageID[0], ID)
```

Comment:
As a good practices, avoid the indexation of an array.
In this case sourceLanguageID could be gotten directly.

```
r.URL.Query().Get("source")
```
That will be `""` if the value does no exist, otherwise will contain the value of the query param.

Reply:
nice, thanks for the recommendation